src/main/java/ch/uzh/ifi/hase/soprafs22/lobby/Lobby.java

```diff
@@ -132,6 +128,43 @@ public void assignNewHost() {
         }
     }
 
+    @Override
+    public void reduceLobbyCapacity() {
+        Iterator <IPlayer> it = iterator();
+        while(it.hasNext()){
+            if(it.next() != host && lobbyCapacity < getNumberOfPlayers()){
+                it.remove();
```

Comment:
Doesn't removing a element, invalidate the iterator?

Reply:
Nop. Java magic! I got the idea from here: https://stackoverflow.com/questions/10431981/remove-elements-from-collection-while-iterating

Reply:
:o, good to know!